Assets/Scripts/Tools/ToolController.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Assets.Scripts.Tools
+{
+  public class ToolController : MonoBehaviour
+  {
+    private ITool _activeTool;
+    private Camera _attachedCamera;
+    private GraphService _graphService;
+    private ISet<IToolChangeObserver> _toolChangeObservers;
+
+    private IDictionary<KeyCode, ITool> _tools;
+    [SerializeField] private Color activatedColor = Color.yellow;
+    [SerializeField] private Image crosshair;
+    [SerializeField] private float hitDistance = 40f;
+    [SerializeField] private Color notActivatedColor = Color.white;
+
+    public ITool ActiveTool
+    {
+      get => _activeTool;
+      set
+      {
+        _activeTool = value;
+        foreach (var observer in _toolChangeObservers)
+        {
+          observer.OnToolsChanged(_activeTool);
+        }
+      }
+    }
+
+    private void Start()
+    {
+      _attachedCamera = Camera.main;
+      _graphService = FindObjectOfType<GraphService>();
+
+      var toolGunController = FindObjectOfType<ToolTextPanelController>();
+      var edgeTool = new EdgeTool(_graphService);
+
```

Comment:
any reason why some tools are created outside and some inside dictionary initialization?